src/h5web/vis-packs/core/line/LineVis.tsx

```diff
@@ -70,7 +70,10 @@ function LineVis(props: Props) {
     );
   }
 
-  const abscissaToIndex = getValueToIndexScale(abscissas, true);
+  const [abscissas, abscissaToIndex] = useMemo(() => {
+    const abscissas = abscissaValue || range(dataArray.size);
+    return [abscissas, getValueToIndexScale(abscissas, true)];
+  }, [abscissaValue, dataArray.size]);
```

Comment:
This constitutes the big optimisation: 

- When no custom abscissa values were passed, we were recreating the `abscissas` range on every render (for instance, when slicing). More importantly, the change to `abscissas` was invalidating the `useMemo`s further down, so the domain was being recomputed on every render as well.
- `getValueToIndexScale`, which is actually quite computationally intensive (it creates a `range` and loops through it), was not wrapped in `useMemo`, so it was also called on ever render.